Backpack/src/main/java/net/skyscanner/backpack/starrating/BpkInteractiveStarRating.kt

```diff
@@ -38,7 +38,10 @@ open class BpkInteractiveStarRating @JvmOverloads constructor(
     val x = if (layoutDirection == View.LAYOUT_DIRECTION_RTL) width - ev.x else ev.x
     val itemWidth = width / maxRating
     val selectedItems = x / itemWidth
-    rating = Math.max(1f, selectedItems + 0.5f)
+    super.rating = Math.round(Math.max(1f, selectedItems + 0.5f)).toFloat()
+    if (ev.action == MotionEvent.ACTION_UP || ev.action == MotionEvent.ACTION_CANCEL) {
```

Comment:
Why do we trigger for `ACTION_CANCEL` too? Seems like `ACTION_CANCEL` should cancel the interaction no?

Reply:
Since the UI updates together with movement, on CANCEL it will remain the its last state. However callback won't be called unless you add this check.

Reply:
👍